### PR TITLE
crypto: use one place to sign in test

### PR DESCRIPTION
**Pull request by joyqvq**
this is to suppress `Signature::new` into one place. this helps to centralize all intent signing related change into one place.

### CODE REVIEW

File: crates/sui-core/src/test_utils.rs [hunk 1]
```diff
@@ -138,6 +138,15 @@ pub fn create_fake_transaction() -> Transaction {
...
+    to_sender_signed_transaction(data, &sender_key)
+}
+
+// This is used to sign transaction with signer using default Intent.
+pub fn to_sender_signed_transaction(
+    data: TransactionData,
+    signer: &dyn Signer<Signature>,
+) -> Transaction {
+    let signature = Signature::new_temp(&data.to_bytes(), signer);
+    // let signature = Signature::new_secure(&data, Intent::default(), signer).unwrap();
```
Comment (joyqvq) [inline crates/sui-core/src/test_utils.rs:150]: this is to demonstrate where the future intent signing logic lives. it should just be a simple switch here to use intent signing for all tests, spec generations

File: crates/sui-types/src/crypto.rs [hunk 2]
```diff
@@ -704,6 +703,24 @@ impl Signature {
         value.write(&mut message);
         secret.sign(&message)
     }
+
+    pub fn new_temp(value: &[u8], secret: &dyn signature::Signer<Signature>) -> Signature {
+        secret.sign(value)
+    }
+
+    // pub fn new_secure<T>(
```
Comment (joyqvq) [inline crates/sui-types/src/crypto.rs:711]: new_temp will be replaced by new_secure. `Signature::new` only works on Signable trait struct that we eventually will not need.

File: crates/sui-core/src/unit_tests/authority_aggregator_tests.rs [hunk 3]
```diff
@@ -295,11 +295,6 @@ pub fn set_object_move_transaction(
     )
 }
 
-pub fn to_transaction(data: TransactionData, signer: &dyn Signer<Signature>) -> Transaction {
-    let signature = Signature::new(&data, signer);
-    Transaction::new(data, signature)
-}
```
Comment (joyqvq) [inline crates/sui-core/src/unit_tests/authority_aggregator_tests.rs:301]: basically make this util function shared by more tests

File: crates/sui-core/src/test_utils.rs [hunk 4]
```diff
@@ -140,6 +140,15 @@ pub fn create_fake_transaction() -> Transaction {
         object.compute_object_reference(),
         10000,
     );
-    let signature = Signature::new(&data, &sender_key);
+    to_sender_signed_transaction(data, &sender_key)
+}
+
+// This is used to sign transaction with signer using default Intent.
```
Comment (kchalkias) [inline crates/sui-core/src/test_utils.rs:146]: I guess this comment will apply in the near future, as don't use intents yet.

**Review by kchalkias — APPROVED** [review]
LGTM, just a small comment

**Pull request merged**